Week3/TODO_API.md

```diff
@@ -0,0 +1,53 @@
+| `GET /user`               | Get all user information        |                                                                                     |                              |
+| `GET /user/:userid`       | Get user information by user id |                                                                                     |                              |
+| `POST /user`              | Create new user                 | `{`<br/>`"username": "John Doe",`<br/>`"email": "johndoe@example.com"`<br/>`}`      | username, email(required)    |
+| `DELETE /user/:userid`    | Delete user by user id          |                                                                                     |                              |
+| `GET /user/:userid/list`  | Get lists of a user by user id  |                                                                                     |                              |
+| `POST /user/:userid/list` | Create new list for a user      | `{`<br/>`"listname": "Shopping List",`<br/>`"alert": "2019-10-01 12:00:00"`<br/>`}` | Set reminder alert(optional) |
+
+---------------
+
+### *`/list` endpoint*
+
+| **Request**                              | **Description**                      | **Body**                                                         | **Details**                             |
+| ---------------------------------------- | ------------------------------------ | ---------------------------------------------------------------- | --------------------------------------- |
+| `GET /list/:listid`                      | Get a list by list id                |                                                                  |                                         |
+| `DELETE /list/:listid`                   | Delete a list by listid              |                                                                  |                                         |
+| `GET /list/:listid/todo`                 | Get all Todos of a list by listid    |                                                                  |                                         |
+| `GET /list/:listid/todo/:todoid`         | Get Todo by todoid                   |                                                                  |                                         |
+| `POST /list/:listid/todo`                | Add new Todo to a list by list id    | `{`<br/>`"note": "Buy milk",`<br/>`"tags": "food,daily"`<br/>`}` | add multiple tags using comma(optional) |
+| `DELETE /list/:listid/todo/:todoid`      | Delete a Todo by todo id             |                                                                  |                                         |
+| `POST /list/:listid/todo/:todoid/done`   | Mark Todo as done by todo id         |                                                                  |                                         |
```

Comment:
Here I would not have use a delete for the done but an other endpoint as done is not an entity
post /list/:listid/todo/:todoid/done = done
post /list/:listid/todo/:todoid/undo = not done

Reply:
Changed the HTTP verb to PATCH since making a partial modify following the [convention](https://www.restapitutorial.com/lessons/httpmethods.html).
`PATCH /list/:listid/todo/:todoid/done` done
`PATCH /list/:listid/todo/:todoid/undo`  not done

Thanks for the tip